### PR TITLE
Luasnip

### PR DESCRIPTION
**Pull request by ray-x**
First commit for [luasnip](https://github.com/L3MON4D3/LuaSnip)
Most of the snip code is from this repo [shark](https://github.com/arsham/shark#snippets)
and TJ's neovim setup [config_manager](https://github.com/tjdevries/config_manager)
and [ziontee113](https://github.com/ziontee113/luasnip-tutorial)

### CODE REVIEW

**Comment by arsham** [comment]
Thanks for your kind complement mate!

File: lua/snips/go.lua [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      ]],
+      { ls.i(1), ls.i(2) }
+    ),
+    in_test_fn
+  ),
+
+  -- Create Mocks
+  ls.s(
+    { trig = "mock", name = "Mocks", dscr = "Create a mock with defering assertion" },
+    fmt("{} := &mocks.{}{{}}\ndefer {}.AssertExpectations(t)\n{}", {
```
Comment (arsham) [inline lua/snips/go.lua:157]: Mockery since v2.11 has a new handy function you might want to use:

```go
factory := mocks.NewFactory(t)
```

**Comment by arsham** [comment]
Have a look at [this](https://github.com/arsham/shark/blob/master/lua/settings/luasnip/go.lua#L235..L243) snippet as well. It helps creating [this](https://github.com/arsham/dbtools/blob/master/db_test.go#L48..L78) style of sub tests.

**Comment by ray-x** [comment]
Your [snippet](https://github.com/arsham/shark/blob/master/lua/settings/luasnip/go.lua#L235..L243)  is a very cool dynamic node usage in testing.  I have brought that in.

**Pull request merged**